tripal/tripal
# Merged data collections with the file management

Issue #555

## Type(s) of Change(s)


## Description


Files created via data collections were not integrated with the file management system and therefore not subject to quotas.  This fixes that problem.  It also forced a number of changes:

1)  The "Data Collections" tab no longer exists on the user's profile page. Instead collections are now dealt with via the 'Files' tab.  See the screen shot.  
2) This updated Files tab provides a table that lists data collections that are "pending" (i.e. files haven't been generated yet). It also provides an interactive file browser.
3) I fixed a bug in that data collections always reported only having 1 record. 
4) Data collection files are now stored in a 'data_collections' directory which makes them easier to spot in the interactive file browser.

## Testing?




1.  Pull this branch
2.  clear your cache (drush cc all)
2.  Create a new data collection using Views.
3.  Go to the files page on your user profile page.  You should see your data collection in the 'Pending Data Collections' table.
4.  Click the 'Create Files' link to generate the files, and run the job.
5. Reload the Files tab. You should now find the files for the data collection in the 'data_collections' folder in the interactive file browser.  
6.  Perform any additional actions: e.g. download the file, renew the expiration date, delete a single file, delete an entire collection, etc.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/1719352/44617648-534def80-a81b-11e8-928d-9625288c6a00.png)

## Additional Notes (if any):
